server/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -2218,8 +2218,13 @@ public void testSnapshotRelocatingPrimary() throws Exception {
                         .put("compress", randomBoolean())
                         .put("chunk_size", randomIntBetween(100, 1000), ByteSizeUnit.BYTES)));
 
-        // Create index on 1 nodes and make sure each node has a primary by setting no replicas
-        assertAcked(prepareCreate("test-idx", 1, Settings.builder().put("number_of_replicas", 0)));
+        internalCluster().ensureAtLeastNumDataNodes(2);
+        // Create index on two nodes and make sure each node has a primary by setting no replicas
+        assertAcked(prepareCreate("test-idx", 2, Settings.builder()
+            .put(SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(SETTING_NUMBER_OF_SHARDS, 2)));
```

Comment:
Before this change I think we had up to 10 shards and therefore would have had more relocations (mod bugs ofc). I think we only get one relocation here. Can we have more shards just in case that helps hit something?

Reply:
Sure, made it use between 2 and 10 shards now